src/lib/npm.ts

```diff
@@ -39,32 +40,22 @@ const getNpmPath = async () => {
   return join(node_modules(), ".bin", "npm");
 };
 
+let npmExecutable: string | undefined;
+
 async function runNpm(args: Omit<ExecPromiseArg, "command">) {
-  let res: string;
+  if (typeof npmExecutable === "string") {
+    return execPromise({ command: npmExecutable, ...args });
+  }
 
   try {
-    res = await execPromise({
-      command: await getNpmPath(),
-      ...args,
-    });
-  } catch (localNpmErr) {
-    try {
-      res = await execPromise({
-        command: "npm",
-        ...args,
-      });
-    } catch (globalNpmErr) {
-      commandObj.error(
-        `Can't find ${color.yellow(
-          "npm",
-        )} in your system. Please make sure it's available on PATH. Errors:\n\n${stringifyUnknown(
-          localNpmErr,
-        )}\n\n${stringifyUnknown(globalNpmErr)}`,
-      );
-    }
+    const npmPath = await getNpmPath();
+    await access(npmPath);
```

Comment:
How will the error look like? Maybe catch the error, and explain what happened? Which paths were tried, which ENVs, and so on.

Reply:
I improved the error message so now it will look like this

```
Error: Not able to install @fluencelabs/registry@0.8.7 to /home/shams/.fluence/npm/@fluencelabs/registry/0.8.7. 
Error: Couldn't find npm executable. Tried at /home/shams/Projects/work/fluence-cli/node_modules/.bin/npm and also tried finding npm in your $PATH. Please make sure npm is available
```